corehq/apps/case_importer/tracking/models.py

```diff
@@ -23,6 +27,29 @@ class Meta(object):
     def task(self):
         return get_task(self.task_id)
 
+    @memoized
+    def get_task_status_json(self):
+        if self.task_status_json:
+            return TaskStatus.wrap(self.task_status_json)
+        else:
+            return get_task_status_json(str(self.task_id))
+
+    def set_task_status_json_if_finished(self):
+        """
+        set task_status_json based on self.task_id
+
+        :return: True if the caller must save model to persist the effect
+
+        """
+        if self.task_status_json is None:
+            # intentionally routing through method to prime local cache
+            task_status_json = self.get_task_status_json()
+            if task_status_json.is_finished():
+                self.task_status_json = task_status_json
+                return True
```

Comment:
just curious, why not just `save` the model here instead of return `True` and then saving?